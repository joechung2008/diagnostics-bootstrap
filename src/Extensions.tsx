import { Nav } from "react-bootstrap";
import { byKey, isExtensionInfo, toNavLink } from "./utils";

const Extensions: React.FC<ExtensionsProps> = ({ extensions, onLinkClick }) => {
  const links = Object.values(extensions)
    .filter(isExtensionInfo)
    .map(toNavLink)
    .sort(byKey);

  return (
    <div className="extension-root overflow-x-hidden">
      <Nav variant="pills" className="flex-column" aria-label="Extensions">
        {links.map((link) => (
          <Nav.Link
            key={link.key}
            onClick={(e) => {
              e.preventDefault();
              onLinkClick?.(e, link);
            }}
          >
            {link.name}
          </Nav.Link>
        ))}
      </Nav>
    </div>
  );
};

export default Extensions;
