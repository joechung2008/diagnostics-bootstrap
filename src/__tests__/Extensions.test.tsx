import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Extensions from "../Extensions";

describe("Extensions", () => {
  it("renders correctly", () => {
    const extensions = {
      ext1: { extensionName: "ext1" },
      ext2: { extensionName: "ext2" },
    };
    const { container } = render(
      <Extensions extensions={extensions} onLinkClick={() => {}} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
