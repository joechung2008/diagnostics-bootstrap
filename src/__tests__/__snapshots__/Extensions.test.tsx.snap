// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extensions > renders correctly 1`] = `
<div
  class="extension-root overflow-x-hidden"
>
  <div
    aria-label="Extensions"
    class="flex-column nav nav-pills"
  >
    <a
      class="nav-link"
      href="#"
      role="button"
      tabindex="0"
    >
      ext1
    </a>
    <a
      class="nav-link"
      href="#"
      role="button"
      tabindex="0"
    >
      ext2
    </a>
  </div>
</div>
`;
